src/DOOM/m_menu.c

```diff
@@ -1418,6 +1418,23 @@ void M_FinishReadThis(int choice)
     M_SetupNextMenu(&MainDef);
 }
 
+//
+// When a secret is found (ColleagueRiley)
+//
+void doom_secretFound(int s) {
+    static lastTime = 0;
+    if (lastTime == 0) {
+        lastTime = I_GetTime();
+        S_StartSound(0, sfx_getpow);
+    }
+    
+    if (I_GetTime() >= lastTime + 20)
+        lastTime = 0;
```

Comment:
Can this message be interrupted by another one? Then you won't reset the countdown.

Reply:
That is a possible concern. I think it should be fine. But if it's not then it should be changed to get the time directly.